### PR TITLE
chore: upgrade hyper to 1.2.0

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Main changes:
- we have to upgrade `axum` as well to use latest hyper
- we have to upgrade `tower-http` as well
- there's no client in `hyper` anymore. `reqwest` does everything we were doing with `hyper` client + it's higher level (no need to worry about https adapter, simpler API etc..). Overall looks the same
- hyper `Body` is now a trait. This problem is also solved by using `reqwest`
- axum order of parameters in handlers is different (json is last now)
- remove use of `Extension` in favor of `State` (which is typesafe unlike `Extension`). `State` is the new preferred way.
- `axum` way to create the server is a tiny bit different, but that's minor

## Risk

inverted 80/20 type of deal

Tested every provider locally

## Deploy Plan

~~Test thoroughly~~

### CODE REVIEW

File: core/src/providers/cohere.rs [hunk 1]
```diff
@@ -434,40 +417,33 @@ impl CohereEmbedder {
...
         assert!(self.api_key.is_some());
 
-        let https = HttpsConnector::new();
-        let cli = Client::builder().build::<_, hyper::Body>(https);
-
-        let req = Request::builder()
-            .method(Method::POST)
-            .uri(self.uri()?)
+        let res = reqwest::Client::new()
+            .post(self.uri()?.to_string())
```
Comment (fontanierh) [inline core/src/providers/cohere.rs:421]: Got a bit lazy here, there's likely a way to impl `IntoUrl` without copying

File: core/bin/sqlite_worker.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use anyhow::{anyhow, Result};
 use axum::{
-    extract,
+    extract::{Path, State},
     response::Json,
```
Comment (fontanierh) [inline core/bin/sqlite_worker.rs:4]: turns out this is both the Response and the Extractor.

File: core/src/providers/azure_openai.rs [hunk 3]
```diff
@@ -51,23 +49,11 @@ struct AzureOpenAIDeployments {
 async fn get_deployments(endpoint: &str, api_key: &str) -> Result<Vec<AzureOpenAIDeployment>> {
     let url = format!("{}openai/deployments?api-version=2022-12-01", endpoint);
 
-    let mut req = Request::builder().method(Method::GET).uri(url.as_str());
-
-    {
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/providers/azure_openai.rs:56]: pleasing the type system

File: core/src/providers/cohere.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/cohere.rs:421]: fine

**Review by spolu — APPROVED** [review]
Looks great

**Pull request merged**